browser/infobars/sync_v1_deprecation_infobar_delegate.cc

```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/infobars/sync_v1_deprecation_infobar_delegate.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/feature_list.h"
+#include "brave/common/pref_names.h"
+#include "brave/common/url_constants.h"
+#include "brave/components/brave_sync/brave_sync_prefs.h"
+#include "brave/components/brave_sync/features.h"
+#include "brave/components/p3a/pref_names.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/infobars/infobar_service.h"
+#include "chrome/grit/chromium_strings.h"
+#include "components/infobars/core/infobar.h"
+#include "components/prefs/pref_service.h"
+#include "components/strings/grit/components_strings.h"
+#include "components/vector_icons/vector_icons.h"
+#include "ui/base/l10n/l10n_util.h"
+
+// static
+void SyncV1DeprecationInfoBarDelegate::Create(InfoBarService* infobar_service,
+                                            PrefService* prefs) {
+  // Don't show infobar if:
+  // - sync v1 flag is not enabled
+  // - sync is disabled (pref, not flag)
+  // - notice has already been dismissed / viewed
+  const bool is_flag_enabled =
+      base::FeatureList::IsEnabled(brave_sync::features::kBraveSync);
+  if (!is_flag_enabled) {
+    return;
+  }
+  if (!prefs->GetBoolean(brave_sync::prefs::kSyncEnabled) ||
+      prefs->GetBoolean(
+          brave_sync::prefs::kSyncDeprecationWarningNoticeDismissed)) {
+    return;
+  }
+
+  // Only show the bar once, ever
```

Comment:
If user uses multiple profile, maybe that user could see this infobar multiple times.

Reply:
I'm ok with that, the other profiles must have sync enabled too. We can't assume that the other profiles are not other people perhaps.